main.py

```diff
@@ -550,21 +555,25 @@ def _send_flow_mods_from_request(self, dpid, command, flows_dict=None):
             f"Send FlowMod from request dpid: {dpid} command: {command}"
             f" flows_dict: {flows_dict}"
         )
-        if dpid:
+        try:
+            if not dpid:
+                self._install_flows(
+                    command, flows_dict, self._get_all_switches_enabled()
+                )
+                return jsonify({"response": "FlowMod Messages Sent"}), 202
+
             switch = self.controller.get_switch_by_dpid(dpid)
             if not switch:
                 return jsonify({"response": "dpid not found."}), 404
-            elif switch.is_enabled() is False:
-                if command == "delete":
-                    self._install_flows(command, flows_dict, [switch])
-                else:
-                    return jsonify({"response": "switch is disabled."}), 404
-            else:
-                self._install_flows(command, flows_dict, [switch])
-        else:
-            self._install_flows(command, flows_dict, self._get_all_switches_enabled())
 
-        return jsonify({"response": "FlowMod Messages Sent"}), 202
+            if not switch.is_enabled() and command == "add":
+                return jsonify({"response": "switch is disabled."}), 404
```

Comment:
As you are already changing this, it is better to raise the exception.

```suggestion
                raise NotFound("switch is disabled.")
```

Reply:
I've address this comment on PR #49 fyi, I've set this one here as draft (same changes there but different base branch, since archive_flows will likely need to wait some DB improvements)